job/cache.go

```diff
@@ -203,27 +203,48 @@ func (c *MemoryJobCache) PersistEvery(persistWaitTime time.Duration) {
 }
 
 type LockFreeJobCache struct {
-	jobs            *hashmap.HashMap
-	jobDB           JobDB
-	retentionPeriod time.Duration
-	PersistOnWrite  bool
+	jobs             *hashmap.HashMap
+	jobDB            JobDB
+	dbCleaning       DBCleaning
+	retentionPeriod  time.Duration
+	durationClearJob time.Duration
+	PersistOnWrite   bool
 	Clock
 }
 
-func NewLockFreeJobCache(jobDB JobDB) *LockFreeJobCache {
-	return &LockFreeJobCache{
-		jobs:            hashmap.New(8), //nolint:gomnd
-		jobDB:           jobDB,
-		retentionPeriod: -1,
+func NewLockFreeJobCache(jobDB JobDB, opts ...LockFreeJobCacheOption) *LockFreeJobCache {
+	l := LockFreeJobCache{
+		jobs:             hashmap.New(8), //nolint:gomnd
+		jobDB:            jobDB,
+		retentionPeriod:  -1,
+		durationClearJob: 5 * time.Minute,
+	}
+	for _, opt := range opts {
+		opt(&l)
+	}
+	return &l
+}
+
+type LockFreeJobCacheOption func(cache *LockFreeJobCache)
+
+func WithDBCleaning(cleaning DBCleaning) LockFreeJobCacheOption {
+	return func(cache *LockFreeJobCache) {
+		cache.dbCleaning = cleaning
 	}
 }
 
 func (c *LockFreeJobCache) Start(persistWaitTime time.Duration, jobstatTtl time.Duration) {
 	if persistWaitTime == 0 {
 		c.PersistOnWrite = true
 	}
+	//Start Cron job Clean job is done
+	go c.cronClearJobDoNotUse()
```

Comment:
Why put this here instead of below the line `247` below to completely eliminate the possibility of race?